File: launchdarkly/python/src/CreateMetricGroupExample.py
import json
from datetime import date, datetime
from pprint import pprint

from launchdarkly_client import ApiClient, ApiException, Configuration, api, models

configuration = Configuration(
    api_key={"ApiKey": "YOUR_API_KEY"},
)

with ApiClient(configuration) as api_client:
    metrics_1 = models.MetricInMetricGroupInput(
        key="metric-key-123abc",
        nameInGroup="Step 1",
    )

    metrics = [
        metrics_1,
    ]

    metric_group_post = models.MetricGroupPost(
        key="metric-group-key-123abc",
        name="My metric group",
        kind="funnel",
        maintainerId="569fdeadbeef1644facecafe",
        tags=[
            "ops",
        ],
        description="Description of the metric group",
        metrics=metrics,
    )

    try:
        response = api.MetricsBetaApi(api_client).create_metric_group(
            project_key=None,
            metric_group_post=metric_group_post,
        )

        pprint(response)
    except ApiException as e:
        print("Exception when calling MetricsBetaApi#create_metric_group: %s\n" % e)

File: launchdarkly/ruby/src/CreateMetricGroupExample.rb
require "json"
require "launchdarkly_client"

LaunchDarklyClient.configure do |config|
    config.api_key["Authorization"] = "YOUR_API_KEY"
end

metrics_1 = LaunchDarklyClient::MetricInMetricGroupInput.new
metrics_1.key = "metric-key-123abc"
metrics_1.name_in_group = "Step 1"

metrics = [
    metrics_1,
]

metric_group_post = LaunchDarklyClient::MetricGroupPost.new
metric_group_post.key = "metric-group-key-123abc"
metric_group_post.name = "My metric group"
metric_group_post.kind = "funnel"
metric_group_post.maintainer_id = "569fdeadbeef1644facecafe"
metric_group_post.tags = [
    "ops",
]
metric_group_post.description = "Description of the metric group"
metric_group_post.metrics = metrics

begin
    response = LaunchDarklyClient::MetricsBetaApi.new.create_metric_group(
        nil, # project_key
        metric_group_post,
    )

    p response
rescue LaunchDarklyClient::ApiError => e
    puts "Exception when calling MetricsBetaApi#create_metric_group: #{e}"
end
